### PR TITLE
haskell.nix for build 

### PR DESCRIPTION
**Pull request by drownbes**

### CODE REVIEW

**Review by tim2CF — CHANGES_REQUESTED** [review]

File: .bitcoin/bitcoin.conf [hunk 1]
```diff
@@ -1,5 +1,4 @@
 regtest=1
-daemon=1
```
[the remark this answers is not in the document]
Comment (drownbes) [reply .bitcoin/bitcoin.conf:2]: i need pid then starting....

File: .dockerignore [hunk 2]
```diff
@@ -0,0 +1,4 @@
+.git
+dist-newstyle/
+result
+result-*
```
[the remark this answers is not in the document]
Comment (drownbes) [reply .dockerignore:4]: it is preference for adding work in docker

File: .envrc [hunk 3]
```diff
@@ -0,0 +1 @@
+use_nix
```
[the remark this answers is not in the document]
Comment (drownbes) [reply .envrc:1]: for direnv. If you don't have its doing nothing

File: .editorconfig [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+
+[*]
+indent_style = space
+indent_size = 2
+end_of_line = lf
+charset = utf-8
+trim_trailing_whitespace = true
+insert_final_newline = true
+max_line_length = 100
+
```
[the remark this answers is not in the document]
Comment (drownbes) [reply .editorconfig:11]: i have no idea, maybe

File: hie.yaml [hunk 5]
```diff
@@ -1,2 +1,2 @@
 cradle:
-  stack:
+  cabal:
```
[the remark this answers is not in the document]
Comment (drownbes) [reply hie.yaml:2]: sometimes it work without that hehe

File: shell.nix [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+    haskellPackages.stack
+    pkgs.bitcoin
+    pkgs.openssl
+    pkgs.expect
+    pkgs.ps
+    lnd
+  ];
+  tools = {
+    cabal = "3.2.0.0";
+    hlint = "3.2.7";
```
[the remark this answers is not in the document]
Comment (drownbes) [reply shell.nix:23]: next time. separate task

File: script/prepare-test-env.sh [hunk 7]
```diff
@@ -0,0 +1,7 @@
+#!/bin/sh
+
+./script/generate-tls-cert.sh
+./script/reset-test-data.sh
+./script/spawn-test-deps.sh
+./script/create-wallets.sh
+
```
Comment (tim2CF) [inline script/prepare-test-env.sh:7]: add `set -e` to all sh files

**Review by tim2CF — APPROVED** [review]

**Review by prushin — APPROVED** [review]

**Pull request merged**